src/fuzzy.js

```diff
@@ -299,8 +299,9 @@ function sellersShouldContinue(node, _, term, threshold, sValue, lastValue) {
 	return 1 - (bestPossibleValue / term.length) >= threshold;
 }
 
-// recursively walk the trie
+// recursively walk the tree
 function searchRecurse(node, acc, len, term, scoreMethods, rows, results, resultMap, options, sIndex, sValue) {
+	if (options.maxRecursions && len > options.maxRecursions) return;
```

Comment:
While this *can* be used to prevent the crash, it doesn't do so in a particularly satisfying way.
An early exit during the recursion will skip the rest of the subtree: any search candidates contained in a subtree deeper than `maxRecursions` will not be returned (unless using sellers, in which case it *might*, but only if the substring match happens at an index < `maxRecursions`. Any matches after that will simply not be found).

Since this essentially means that some input strings are simply too big, a perhaps more appropriate solution would be to throw an error if a candidate is too big.